hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/UpdateProcessorSupplier.java

```diff
@@ -0,0 +1,149 @@
+import org.bson.conversions.Bson;
+import org.bson.types.ObjectId;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static com.hazelcast.jet.mongodb.MongoDBSinkBuilder.DEFAULT_COMMIT_RETRY_STRATEGY;
+import static com.hazelcast.jet.mongodb.MongoDBSinkBuilder.DEFAULT_TRANSACTION_OPTION;
+import static com.hazelcast.jet.mongodb.impl.Mappers.defaultCodecRegistry;
+import static java.util.Arrays.asList;
+
+/**
+ * ProcessorSupplier that creates {@linkplain WriteMongoP} processors on each instance,
+ *  that will update given items.
+ */
+public class UpdateProcessorSupplier implements ProcessorSupplier {
+
+    private final String connectionString;
+    private final String databaseName;
+    private final String collectionName;
+    private final List<String> fieldNames;
+    private final List<Object> updates;
+    private final List<String> pkFields;
+    private ExpressionEvalContext evalContext;
+
+    UpdateProcessorSupplier(MongoTable table, List<String> fieldNames, List<Object> updates) {
+        this.connectionString = table.connectionString;
+        this.databaseName = table.databaseName;
+        this.collectionName = table.collectionName;
+
+        // update-specific
+        this.fieldNames = fieldNames;
+        this.updates = updates;
+        this.pkFields = Collections.singletonList("_id");
+    }
+
+    @Override
+    public void init(@Nonnull Context context) throws Exception {
+        evalContext = ExpressionEvalContext.from(context);
+    }
+
+    @Nonnull
+    @Override
+    public Collection<? extends Processor> get(int count) {
+        Processor[] processors = new Processor[count];
+
+        for (int i = 0; i < count; i++) {
+            Processor processor = new WriteMongoP<>(
+                    new WriteMongoParams<Document>()
+                            .setClientSupplier(() -> MongoClients.create(connectionString))
+                            .setDatabaseName(databaseName)
+                            .setCollectionName(collectionName)
+                            .setDocumentType(Document.class)
+                            .setDocumentIdentityFn(doc -> doc.getObjectId("_id"))
+                            .setDocumentIdentityFieldName("_id")
+                            .setCommitRetryStrategy(DEFAULT_COMMIT_RETRY_STRATEGY)
+                            .setTransactionOptionsSup(() -> DEFAULT_TRANSACTION_OPTION)
+                            .setIntermediateMappingFn(this::rowToUpdateDoc)
+                            .setWriteMode(WriteMode.UPDATE_ONLY)
+                            .setWriteModelFn(this::write)
+            );
+
+            processors[i] = processor;
+        }
+        return asList(processors);
+    }
+
+    private WriteModel<Document> write(Document row) {
+        List<Bson> updateToPerform = new ArrayList<>();
+        for (int i = 0; i < fieldNames.size(); i++) {
```

Comment:
You can iterate with 
`for (String fieldName : fieldNames) `

Reply:
Yes, but I also need to get corresponding update expression - so by-index is easier